sdk/storage/azure-storage-files-datalake/test/ut/datalake_file_system_client_test.cpp

```diff
@@ -397,6 +398,133 @@ namespace Azure { namespace Storage { namespace Test {
     }
   }
 
+  TEST_F(DataLakeFileSystemClientTest, CustomerProvidedKey_LIVEONLY_)
+  {
+    // will skip test under some cased where test can't run (usually LIVE only tests)
+    CHECK_SKIP_TEST();
+
+    auto getRandomCustomerProvidedKey = [&]() {
+      Files::DataLake::EncryptionKey key;
+      std::vector<uint8_t> aes256Key;
+      aes256Key.resize(32);
+      RandomBuffer(&aes256Key[0], aes256Key.size());
+      key.Key = Azure::Core::Convert::Base64Encode(aes256Key);
+      key.KeyHash = Azure::Core::Cryptography::_internal::Sha256Hash().Final(
+          aes256Key.data(), aes256Key.size());
+      key.Algorithm = Blobs::Models::EncryptionAlgorithmType::Aes256;
+      return key;
+    };
+
+    const int32_t bufferSize = 4 * 1024; // 4KB data size
+    std::vector<uint8_t> buffer(bufferSize, 'x');
+    Azure::Core::IO::MemoryBodyStream bodyStream(buffer.data(), buffer.size());
+
+    auto customerProvidedKey = getRandomCustomerProvidedKey();
+    Files::DataLake::DataLakeClientOptions options;
+    options.CustomerProvidedKey = customerProvidedKey;
+    auto fileServiceClient
+        = Azure::Storage::Files::DataLake::DataLakeServiceClient::CreateFromConnectionString(
+            AdlsGen2ConnectionString(), options);
+    auto fileSystemClient = fileServiceClient.GetFileSystemClient(m_fileSystemName);
+
```

Comment:
Need to verify CPK option can be inherited when getting filesystem client from service client.

Also need to verify CPK option is correctly set when constructing pathclient, fileclient, directoryclient from construcots (factory methods).